test/k8s/conn/auth/azure_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule K8s.Conn.Auth.AzureTest do
+  @moduledoc false
+  use ExUnit.Case, async: true
+
+  alias K8s.Conn
+  alias K8s.Conn.Auth.Azure
+
+  describe "create/2" do
+    test "creates a Azure struct from  data" do
+      non_expired_unix_ts = DateTime.utc_now() |> DateTime.add(10, :minute) |> DateTime.to_unix()
+
+      auth = %{
+        "auth-provider" => %{
+          "config" => %{
+            "access-token" => "xxx",
+            "apiserver-id" => "service_id",
+            "client-id" => "client_id",
+            "expires-on" => "#{non_expired_unix_ts}",
+            "refresh-token" => "yyy",
+            "tenant-id" => "tenant"
+          },
+          "name" => "azure"
+        }
+      }
+
+      assert {:ok,
+              %Azure{
+                token: "xxx"
+              }} = Azure.create(auth, nil)
+    end
+
+    @tag :skip
```

Comment:
or remove?

Reply:
I added it mostly to demonstrate that we can indeed make a HTTP request against the azure api. I will remove it, not sure we will get much from mocking this out.